salt/utils/sanitizers.py

```diff
@@ -62,3 +63,34 @@ def hostname(value):
 
 
 clean = InputSanitizer()
+
+
+def mask_args_value(data, mask):
+    '''
+    Mask a line in the data, which matches "mask".
+
+    In case you want to put to the logs rosters or other data,
+    but you certainly do not want to put there an actual IP address,
+    passwords, user names etc.
+
+    Note, this is working only when data is a single string,
+    ready for print or dump to the log. Also, when the data is formatted
+    as "key: value" in YAML syntax.
+
+    :param data: String data, already rendered.
+    :param mask: Mask that matches a single line
+
+    :return:
+    '''
+    if not mask:
+        return data
+
+    out = []
+    for line in data.split(os.linesep):
+        if fnmatch.fnmatch(line.strip(), mask) and ':' in line:
+            key, value = line.split(':', 1)
+            out.append('{}: {}'.format(key.strip(), '** hidden **'))
```

Comment:
Just as a precaution, `key.strip()` should probably be `salt.utils.stringutils.to_unicode(key.strip())`, to prevent a `UnicodeDecodeError` if `key` happens to be a `str` type with non-ascii unicode in it.